src/Tools/Common/Commands/Utils.cs

```diff
@@ -59,66 +69,66 @@ public static bool ValidateArgumentsForChildProcess(int processId, string name,
         }
 
         /// <summary>
-        /// A helper method for validating --process-id, --name, --diagnostic-port options for collect commands.
+        /// A helper method for validating --process-id, --name, --diagnostic-port, --dsrouter options for collect commands.
         /// Only one of these options can be specified, so it checks for duplicate options specified and if there is
         /// such duplication, it prints the appropriate error message.
         /// </summary>
         /// <param name="processId">process ID</param>
         /// <param name="name">name</param>
         /// <param name="port">port</param>
+        /// <param name="dsrouter">dsrouter</param>
         /// <param name="resolvedProcessId">resolvedProcessId</param>
         /// <returns></returns>
-        public static bool ValidateArgumentsForAttach(int processId, string name, string port, out int resolvedProcessId)
+        public static bool ValidateArgumentsForAttach(int processId, string name, string port, string dsrouter, out int resolvedProcessId)
```

Comment:
Nit: Lets change the name of this method. If we are launching helper tools that is a big side-effect. 'Validate' makes it sound like the method has no side-effects.

Reply:
ResolveProcessForAttach?